app/src/main/java/com/github/se/polyfit/model/ingredient/Ingredient.kt

```diff
@@ -0,0 +1,74 @@
+package com.github.se.polyfit.model.ingredient
+
+import com.github.se.polyfit.model.meal.Meal
+import kotlin.properties.Delegates
+
+/**
+ * Represents an ingredient in a meal. Automatically updates the parent meal when any property
+ * changes.
+ *
+ * @property name is unique for each meal
+ */
+class Ingredient(
```

Comment:
we might want to abstract this into two classes

an ingredient is something that is "normalized" into base unit, so if its grams it will be 1g, or something similar. Then, we have a second class called serving, each serving will have an ingredient and a serving size for example. This way we can reuse ingredients rather than create a new object each time for the same food. it'll save space because the ingredient object has a lot of fields, but a serving object doesn't need as much